data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java

```diff
@@ -107,13 +107,25 @@ public HttpRequestBuilder withURL(String url) throws DataClientException {
         return this;
     }
 
-    public HttpRequest build() {
+    public HttpRequest build() throws DataClientException {
+        // If has authorization header, ensure it is not sent over insecure channel
+        boolean hasAuth = request.getHeaders().stream().anyMatch(h -> h.getName().equalsIgnoreCase(HttpHeaderName.AUTHORIZATION.toString()));
+        if (hasAuth) {
+            URL url = request.getUrl();
+            boolean isHttp = url.getProtocol().equalsIgnoreCase("http");
+            boolean isLocalhost = url.getHost().toLowerCase().startsWith(CloudInfo.LOCALHOST);
+
+            if (isHttp && !isLocalhost) {
+                throw new DataClientException(url.toString(), "Cannot forward security token to a remote service over insecure " +
```

Comment:
insecure means someone who is not confident in himself.
Unsecure, not secure, unenceypted, etc.